File: sass-ui/src/types/roleImport.ts
// Role import row from Excel
export interface RoleImportRow {
  row_number: number;
  name: string;
  display_name: string;
  description: string;
  is_default: boolean;
  is_active: boolean;
}

// Row-level validation error (reuse from userImport)
export interface RowError {
  row_number: number;
  field: string;
  message: string;
  value?: string;
}

// Validate import request
export interface ValidateRoleImportRequest {
  rows: RoleImportRow[];
}

// Validate import response
export interface ValidateRoleImportResponse {
  is_valid: boolean;
  total_rows: number;
  valid_rows: number;
  error_count: number;
  errors: RowError[];
}

// Bulk import request
export interface BulkRoleImportRequest {
  rows: RoleImportRow[];
  update_mode: 'skip' | 'update';
}

// Import row result
export interface ImportRowResult {
  row_number: number;
  username: string; // Used for role name
  status: 'created' | 'updated' | 'skipped' | 'error';
  message?: string;
}

// Bulk import response
export interface BulkRoleImportResponse {
  total_rows: number;
  created: number;
  updated: number;
  skipped: number;
  failed: number;
  results: ImportRowResult[];
}

// Import step type for multi-step modal
export type ImportStep = 'upload' | 'preview' | 'validating' | 'validation_result' | 'importing' | 'import_result';

File: sass-api/internal/domain/models/role_import.go
package models

// RoleExportRow represents a role for export
type RoleExportRow struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	DisplayName string `json:"display_name"`
	Description string `json:"description"`
	IsDefault   bool   `json:"is_default"`
	IsActive    bool   `json:"is_active"`
}

// ExportRolesResponse represents the response for role export
type ExportRolesResponse struct {
	Roles []RoleExportRow `json:"roles"`
	Total int             `json:"total"`
}

// RoleImportRow represents a role row from import
type RoleImportRow struct {
	RowNumber   int    `json:"row_number"`
	Name        string `json:"name"`
	DisplayName string `json:"display_name"`
	Description string `json:"description"`
	IsDefault   bool   `json:"is_default"`
	IsActive    bool   `json:"is_active"`
}

// ValidateRoleImportRequest represents the request to validate role import data
type ValidateRoleImportRequest struct {
	Rows []RoleImportRow `json:"rows" validate:"required,min=1,dive"`
}

// ValidateRoleImportResponse represents the response for role import validation
type ValidateRoleImportResponse struct {
	IsValid    bool       `json:"is_valid"`
	TotalRows  int        `json:"total_rows"`
	ValidRows  int        `json:"valid_rows"`
	ErrorCount int        `json:"error_count"`
	Errors     []RowError `json:"errors"`
}

// BulkRoleImportRequest represents the request to bulk import roles
type BulkRoleImportRequest struct {
	Rows       []RoleImportRow `json:"rows" validate:"required,min=1,dive"`
	UpdateMode string          `json:"update_mode" validate:"required,oneof=skip update"`
}

// BulkRoleImportResponse represents the response for bulk role import
type BulkRoleImportResponse struct {
	TotalRows int               `json:"total_rows"`
	Created   int               `json:"created"`
	Updated   int               `json:"updated"`
	Skipped   int               `json:"skipped"`
	Failed    int               `json:"failed"`
	Results   []ImportRowResult `json:"results"`
}
